types/treeify/index.d.ts

```diff
@@ -1,28 +1,24 @@
-// Type definitions for treeify 1.0
+// Type definitions for treeify 1.1
 // Project: https://github.com/notatestuser/treeify
 // Definitions by: Mike North <https://github.com/mike-north>
+//                 Chris Arnesen <https://github.com/carnesen>
```

Comment:
As a co-maintainer of these types with me, are you ok with the idea of using the documentation and test-suite as the sole source of truth for the library's public API?

Reply:
I'm not sure I understand the question, but I'd argue that the library's source code is the ultimate source of truth for its public API. Imagine for example that the author of the library converted the source code to TypeScript and ran `tsc` with `declarations` set to true. What would the `.d.ts` files look like? That is how I tend to approach `@types`. In this case, because of this line:

https://github.com/notatestuser/treeify/blob/master/treeify.js#L71

sometimes asTree can return `undefined` therefore the type should include `undefined` in its ReturnType.

Reply:
In this case, I can see evidence in the test suite that returning `undefined` is what the author intends. However, when maintaining these `@types/*` packages, it's often not quite so clear. Adding your name to the list of maintainers means you will get your changes merged in w/o a review, so i want to ensure we're alignment beyond this one code change.

If I had the following code
```js
function combine(a, b) {  return a + b; }
```
and my readme consisted of 
```js
add(3, 4); // 7
```
it sounds like you'd want to add types that support things like 
```js
add('foo ', 'bar'); // "foo bar"
add('foo ', 6); // "foo 6"
add(6, ' bar'); // "6 bar"
```
Am I misrepresenting your view here?

Reply:
Yeah I suppose I would! Absent other information I'd say concatenation is a reasonable use of the library that you might expect to see "in the wild". That's my other consideration is thinking about the person who is using the library in JavaScript and now decides to upgrade to TypeScript. Do the types cover their reasonable use case?